mkosi/installer/rpm.py

```diff
@@ -52,16 +54,23 @@ def find_rpm_gpgkey(
 ) -> Optional[str]:
     # We assume here that GPG keys will only ever be relative symlinks and never absolute symlinks.
 
-    if gpgpath := next((context.config.tools() / "usr/share/distribution-gpg-keys").rglob(key), None):
-        return (Path("/") / gpgpath.resolve().relative_to(context.config.tools())).as_uri()
+    globs = [
+        f"/usr/share/distribution-gpg-keys/**/{key}",
+        f"/etc/pki/rpm-gpg/**/{key}",
+    ]
 
-    # ToolsTreeCertificates= only applies to certificates but the rpm gpg keys in /etc are located within the
-    # /etc/pki certificates directory so as a result the option has to apply to the rpm gpg keys in /etc as
-    # well
-    root = context.config.tools() if context.config.tools_tree_certificates else Path("/")
+    paths = (
+        run(
+            ["bash", "-c", rf"shopt -s nullglob && printf '%s\n' {' '.join(globs)}"],
```

Comment:
That construction did surprise me a bit a first, relying on the glob to be expanded and the expanded thing passed to `printf`. I think some `find /usr/share/distribution-gpg-keys /etc/pki -name <key> -type f` would be easier and we have it in the tools tree.

Reply:
We have it in the tools tree but it's part of the misc profile which is not required to put in the dependency list of mkosi distro packages, so I'd prefer to use bash here to avoid that dependency.